packages/studio/server/src/studio/nlu/nlu-service.ts

```diff
@@ -148,29 +149,26 @@ export class NLUService {
   }
 
   public async downloadAndSaveModelWeights(botId: string) {
-    // TODO: reimpl this
-    /*
     const bot = this._bots[botId]
     if (!bot) {
       throw new BotNotMountedError(botId)
     }
 
-    const botConfig = await this.configProvider.getBotConfig(botId)
+    const botConfig: BotConfig = await Instance.readFile('bot.config.json').then((buf) => JSON.parse(buf.toString()))
 
     if (!botConfig.nluModels) {
       throw new Error('Missing NLU models. Bot is not trained.')
     }
 
     const modelsFolder = 'models'
-    await this.ghost.forBot(botId).deleteFolder(modelsFolder)
+    await Instance.deleteDir(modelsFolder)
 
     for (const lang of Object.keys(botConfig.nluModels)) {
       const modelId = botConfig.nluModels[lang]
       const modelWeights = await bot.downloadModelWeights(botId, modelId)
 
-      await this.ghost.forBot(botId).upsertFile(modelsFolder, `${modelId}.model`, modelWeights)
+      await Instance.upsertFile(path.join(modelsFolder, `${modelId}.model`), modelWeights)
```

Comment:
**question**: have you verified modelWeights encoding ? i seems you save it as raw bytes (or downloadModelWeights already pareses it). I imagine you tried to upload it on cloud and works.

Reply:
I haven't checked the encoding, but I have tried to upload it and it worked